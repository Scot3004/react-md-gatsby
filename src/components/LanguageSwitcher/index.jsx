import React, { Component } from "react";
import { ListItem, MenuButton } from "react-md";
import { translate } from "react-i18next";

class LanguageSwitcher extends Component {
  constructor(props) {
    super(props);
    const { i18n } = this.props;
    this.state = { language: i18n.language };
    this.handleChangeLanguage = this.handleChangeLanguage.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ language: nextProps.i18n.language });
  }

  handleChangeLanguage(lng) {
    const { i18n } = this.props;
    i18n.changeLanguage(lng);
  }

  renderLanguageChoice({ code, label }) {
    return (
      <ListItem
        key={code}
        onClick={() => this.handleChangeLanguage(code)}
        primaryText={`${label} (${code})`}
      />
    );
  }

  render() {
    const languages = [
      { code: "en", label: "English" },
      { code: "es", label: "Espa√±ol" }
    ];

    return (
      <div className="LanguageSwitcher">
        <MenuButton
          flat
          id="lang-menu"
          centered
          menuItems={languages.map(language =>
            this.renderLanguageChoice(language)
          )}
        >
          {this.state.language}
        </MenuButton>
      </div>
    );
  }
}

export default translate()(LanguageSwitcher);
