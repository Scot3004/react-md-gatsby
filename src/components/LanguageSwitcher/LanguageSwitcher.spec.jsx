import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ListItem } from "react-md";
import LanguageSwitcher from './LanguageSwitcher';

jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate HoC receive the t function as a prop
  translate: () => Component => {
    // eslint-disable-next-line no-param-reassign
    Component.defaultProps = { ...Component.defaultProps, t: (text) => text };
    return Component;
  },
}));

const i18n = {
  language: 'es',
  changeLanguage: (lang) => (lang)
}

const spy = jest.spyOn(i18n, 'changeLanguage');

describe('LanguageSwitcher', () => {
  it('should render correct content', () => {
    const tree = shallow(<LanguageSwitcher i18n={i18n} />);
    expect(toJson(tree)).toMatchSnapshot();
    expect(tree.state().language).toBe('es');
  });

  it('should render correct lang choice', () => {
    const tree = shallow(<LanguageSwitcher i18n={i18n} />);
    const lang = shallow(<div>{tree.instance().renderLanguageChoice({code: 'es', label: 'Espa√±ol'})}</div>)
    expect(toJson(lang)).toMatchSnapshot();
    lang.find(ListItem).simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});
