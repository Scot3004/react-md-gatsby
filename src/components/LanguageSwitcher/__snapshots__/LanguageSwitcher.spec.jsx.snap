// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSwitcher should render correct content 1`] = `
<div
  className="LanguageSwitcher"
>
  <MenuButton
    centered={true}
    defaultVisible={false}
    flat={true}
    id="lang-menu"
    menuItems={
      Array [
        <ListItem
          activeClassName="md-text--theme-primary"
          animateNestedItems={true}
          component="div"
          expanderIcon={
            <FontIcon
              iconClassName="material-icons"
            >
              keyboard_arrow_down
            </FontIcon>
          }
          itemComponent="li"
          onClick={[Function]}
          primaryText="English (en)"
        />,
        <ListItem
          activeClassName="md-text--theme-primary"
          animateNestedItems={true}
          component="div"
          expanderIcon={
            <FontIcon
              iconClassName="material-icons"
            >
              keyboard_arrow_down
            </FontIcon>
          }
          itemComponent="li"
          onClick={[Function]}
          primaryText="Español (es)"
        />,
      ]
    }
    repositionOnResize={false}
    repositionOnScroll={true}
  >
    es
  </MenuButton>
</div>
`;

exports[`LanguageSwitcher should render correct lang choice 1`] = `
<div>
  <ListItem
    activeClassName="md-text--theme-primary"
    animateNestedItems={true}
    component="div"
    expanderIcon={
      <FontIcon
        iconClassName="material-icons"
      >
        keyboard_arrow_down
      </FontIcon>
    }
    itemComponent="li"
    key="es"
    onClick={[Function]}
    primaryText="Español (es)"
  />
</div>
`;
